src/test/java/uk/gov/digital/ho/pttg/application/HmrcClientTest.java

```diff
@@ -1,39 +1,64 @@
 package uk.gov.digital.ho.pttg.application;
 
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.LoggingEvent;
+import ch.qos.logback.core.Appender;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import net.logstash.logback.marker.ObjectAppendingMarker;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
-import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.mockito.quality.Strictness;
+import org.slf4j.LoggerFactory;
 import org.springframework.hateoas.Link;
+import uk.gov.digital.ho.pttg.api.RequestHeaderData;
 import uk.gov.digital.ho.pttg.application.domain.Individual;
 
 import java.time.LocalDate;
 import java.time.Month;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.BDDMockito.given;
 import static org.mockito.BDDMockito.then;
 import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.never;
 import static uk.gov.digital.ho.pttg.application.HmrcClientFunctions.getTaxYear;
+import static uk.gov.digital.ho.pttg.application.LogEvent.EVENT;
+import static uk.gov.digital.ho.pttg.application.LogEvent.HMRC_CALL_SKIPPED_SMOKE_TEST;
 
-@RunWith(MockitoJUnitRunner.class)
 public class HmrcClientTest {
 
     private static final LocalDate DEFAULT_PAYE_EPOCH = LocalDate.of(2013, Month.MARCH, 31);
+    private static final LocalDate ANY_DATE = LocalDate.now();
 
     @Mock private Link anyLink;
     @Mock private Individual anyIndividual;
     @Mock private HmrcHateoasClient mockHmrcHateoasClient;
     @Mock private IncomeSummaryContext mockIncomeSummaryContext;
+    @Mock private RequestHeaderData mockRequestHeaderData;
+    @Mock private Appender<ILoggingEvent> mockLogAppender;
 
     private HmrcClient hmrcClient;
 
+    @Rule
+    public MockitoRule unnecessaryStubsRule = MockitoJUnit.rule().strictness(Strictness.STRICT_STUBS);
+
```

Comment:
I thought the existing behaviour we've experienced with Mockito amounts to STRICT_STUBS - but the documentation suggests otherwise.  Would appreciate a chat about this at some point.

Reply:
It does. The MockitoJUnitRunner sets STRICT_STUBS. That's what we usually want but I needed to set the stubbing to LENIENT for one test and the only way to change it between tests is to stop using the MockitoJUnitRunner and instead use a Rule like this. Happy to discuss though if that doesn't make sense.